fails to build on aarch64 (I think)

```
error[E0308]: mismatched types
   --> src/nixstore.rs:241:47
    |
241 |     unsafe { nix_export_path(c_path.as_ptr(), res.as_mut_ptr() as *mut i8, size) };
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `harmonia` due to previous error
builder for '/nix/store/j6x9d1l485xnikd4c697wyg1mzgb1pxc-harmonia-0.1.0.drv' failed with exit code 101
```

--- a/lib/nix.cpp
+++ b/lib/nix.cpp
@@ -237,8 +237,8 @@ struct CBufSink : nix::Sink {
   }
 };
 
-void nix_export_path(const char *path, char *buffer, size_t size) {
-  CBufSink sink(buffer, size);
+void nix_export_path(const char *path, uint8_t *buffer, size_t size) {
+  CBufSink sink((char *)buffer, size);
   store()->exportPath(store()->parseStorePath(path), sink);
 }
 

--- a/lib/nix.h
+++ b/lib/nix.h
@@ -102,7 +102,7 @@ const char *nix_follow_links_to_store_path(const char *path);
 /**
  *
  */
-void nix_export_path(const char *path, char *buffer, size_t size);
+void nix_export_path(const char *path, uint8_t *buffer, size_t size);
 
 /**
  *

--- a/src/nixstore.rs
+++ b/src/nixstore.rs
@@ -4,7 +4,7 @@
 
 use std::collections::HashMap;
 use std::ffi::CStr;
-use std::os::raw::{c_char, c_ulong, c_void};
+use std::os::raw::{c_char, c_uchar, c_ulong, c_void};
 
 #[repr(C)]
 struct NixStrArray {
@@ -50,7 +50,7 @@ extern "C" {
     fn nix_set_verbosity(level: i32);
     fn nix_is_valid_path(path: *const c_char) -> bool;
 
-    fn nix_export_path(path: *const c_char, buf: *mut c_char, size: usize);
+    fn nix_export_path(path: *const c_char, buf: *mut c_uchar, size: usize);
 
     fn nix_query_path_info(path: *const c_char, base32: bool) -> *const NixPathInfo;
 
@@ -238,7 +238,7 @@ pub fn export_path(path: &str, size: usize) -> Option<Vec<u8>> {
     let c_path = c_path.unwrap();
 
     let mut res: Vec<u8> = vec![0; size];
-    unsafe { nix_export_path(c_path.as_ptr(), res.as_mut_ptr() as *mut i8, size) };
+    unsafe { nix_export_path(c_path.as_ptr(), res.as_mut_ptr(), size) };
     Some(res)
 }
 

